lib/elixir/lib/calendar/time.ex

```diff
@@ -98,37 +98,46 @@ defmodule Time do
       {:error, :invalid_time}
       iex> Time.new(23, 59, 60, 999_999)
       {:error, :invalid_time}
+
+      # Invalid microsecond
       iex> Time.new(23, 59, 59, 1_000_000)
-      {:error, :invalid_time}
+      ** (FunctionClauseError) no function clause matching in Time.new/5
 
-      # Invalid precision
-      Time.new(23, 59, 59, {999_999, 10})
-      {:error, :invalid_time}
+      iex> Time.new(23, 59, 59, {999_999, 10})
+      ** (FunctionClauseError) no function clause matching in Time.new/5
```

Comment:
Please revert. It is backwards incompatible and inconsistent with the things like invalid seconds, which do not raise, but return a tuple.

Reply:
reverted